core/src/main/java/org/jdbi/v3/core/argument/PrimitivesArgumentFactory.java

```diff
@@ -13,10 +13,27 @@
  */
 package org.jdbi.v3.core.argument;
 
+import java.lang.reflect.Type;
 import java.sql.PreparedStatement;
 import java.sql.Types;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 
+import org.jdbi.v3.core.argument.internal.StatementBinder;
+import org.jdbi.v3.core.argument.internal.strategies.LoggableBinderArgument;
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+
+// TODO this factory isn't actually preparable anymore
```

Comment:
We should fix this concern before merging.

Reply:
Same as other comment, how can it be preparable if the arguments are dynamic?